File: frontend/src/types/match.ts
export interface MatchScore {
  location_score: number
  time_score: number
  cluster_score: number
  vector_score: number
  final_score: number
}

export interface Childminder {
  id: number
  name: string
  latitude: number
  longitude: number
  experience_years: number
}

export interface Match {
  id: number
  request_id: number
  childminder: Childminder
  score: MatchScore
  explanation: string
}


export interface MatchResultItem {
  childminder_id: number;
  childminder_name: string;
  location_score: number;
  time_score: number;
  cluster_score: number;
  vector_score: number;
  final_score: number;
  rank_position: number;
  explanation: string;
}

export interface MatchResultsResponse {
  request_id: number;
  generated_at: string;
  results: MatchResultItem[];
}

File: backend/app/schemas/match.py
from datetime import datetime
from pydantic import BaseModel


class MatchResultItem(BaseModel):
    childminder_id: int
    childminder_name: str
    location_score: float
    time_score: float
    cluster_score: float
    vector_score: float
    final_score: float
    rank_position: int
    explanation: str


class MatchResultResponse(BaseModel):
    request_id: int
    generated_at: datetime
    results: list[MatchResultItem]